### PR TITLE
multiplicity(n, 0) raises error for any `n`

### PR DESCRIPTION
**Pull request by gxyd**
fixes issue #8159 and #9066

multiplicity(n, 0) used to go in a endless recursion. It now raises an `ValueError`
log(0, n) now returns `zoo`

### CODE REVIEW

**Comment by toolforger** [comment]
Overall design looks good.
Haven't checked implementation details.

File: sympy/functions/elementary/tests/test_exponential.py [hunk 1]
```diff
@@ -167,6 +167,8 @@ def test_log_values():
 
     assert log(oo*I) == oo
     assert log(-oo*I) == oo
+    assert log(0, 2) == zoo
+    assert log(0, 5) == zoo
```
Comment (toolforger) [inline sympy/functions/elementary/tests/test_exponential.py]: Is this related to multiplicity? I'd have thought that this is simply what you return for "it's a complex pole around zero".

Comment (gxyd) [inline sympy/functions/elementary/tests/test_exponential.py]: I should mention the issue number #9066 , regarding log(0, n).

**Comment by jksuom** [comment]
This looks good. Only, I would like to change the message in ValueError. (The user may have a different name for the argument ``n``.) It could also be useful to mention the ``log(0, n)`` issue in the commit message.

**Comment by Sumith1896** [comment]
@gxyd Looking at your comment in https://github.com/sympy/sympy/pull/9069, I assume you have looked at it. Have a look at https://github.com/sympy/sympy/pull/8160 too.
These were previous works on this

**Comment by jksuom** [comment]
> The latter has some other points too.

Is there something that you would specifically like to point out?

**Comment by gxyd** [comment]
@Sumith1896 thanks for mentioning that. I already looked into it.
@jksuom i think one point that is to be looked is definition of `multiplicity` from your point of view. I find the current definition well valid though.

**Comment by jksuom** [comment]
> definition of multiplicity from your point of view

Probably my point of view would not be very much different from that of the others. In more detail, I think that ``multiplicity(p, n)`` should be defined only for ``n != 0``. In addition, I would expect it to be generally useful only for primes ``p``, because in that case we have the equality
``multiplicity(p, n*m) = multiplicity(p, n) + multiplicity(p, m)``.
(I would not object to defining ``multiplicity(p, 0) == oo`` if that should prove useful in some connection, though I doubt it.. But ``log(0, n)`` should then be reconsidered, of course.)

**Comment by jksuom** [comment]
Merging in 24 hours if no-one objects.

**Comment by gxyd** [comment]
Thanks.

**Comment by gxyd** [comment]
@jksuom  Is it ok after the merging to close the related issues ?

**Comment by jksuom** [comment]
I think so. Actually github often does that [automatically] (https://help.github.com/articles/closing-issues-via-commit-messages/). It is possible that the word issue in the commit message has been confusing.
